tools/version.rb

```diff
@@ -117,35 +113,24 @@ def valid_semantic_version?(version)
       puts pkg.name.ljust(35) + 'fake'.lightred if verbose
       next
     end
-    # No upstream update available
-    if pkg.no_upstream_update?
-      puts pkg.name.ljust(35) + 'noupdate'.lightred if verbose
-      next
-    end
     # Some packages don't work with this yet, so gracefully exit now rather than throwing false positives
     upstream_version = get_version(pkg.name.tr('_', '-'), pkg.homepage)
     if upstream_version.nil?
       puts pkg.name.ljust(35) + 'notfound'.lightred if verbose
       next
     end
 
-    status = if upstream_version == pkg.version
-               'uptodate'.ljust(20).lightgreen
-             elsif upstream_version.empty? || pkg.version.empty?
-               'notfound'.ljust(20).lightred
-             elsif valid_semantic_version?(upstream_version) && valid_semantic_version?(pkg.version)
-               if Gem::Version.correct?(upstream_version) && Gem::Version.correct?(pkg.version)
-                 if Gem::Version.new(upstream_version) > Gem::Version.new(pkg.version)
-                   'outdated'.ljust(20).yellow
-                 else
-                   'mismatch'.ljust(20).orange
-                 end
-               else
-                 'notvalid'.ljust(20).lightred
-               end
-             else
-               'notvalid'.ljust(20).lightred
-             end
-    puts pkg.name.ljust(35) + status + pkg.version.ljust(20) + upstream_version unless !verbose && status == 'uptodate'.ljust(20).lightgreen
+    # Bail out if we arent verbose and so dont want to print packages that are up to date.
+    next if Libversion.version_compare2(pkg.version, upstream_version) >= 0 && !verbose
+    # Print the package name.
+    print pkg.name.ljust(35)
+    # Print the package update status.
+    if Libversion.version_compare2(pkg.version, upstream_version) >= 0
+      print 'uptodate'.ljust(20).lightgreen
+    elsif Libversion.version_compare2(pkg.version, upstream_version) == -1
+      print 'outdated'.ljust(20).yellow
+    end
```

Comment:
What do you do in cases where the updated version is less than the current version.  Not just outdated in those cases.

Reply:
As per the PR description:
```
We now have a few false positives because we're accepting higher version levels as uptodate rather than a mismatch, but this is correct in the vast majority of cases and is the best choice.
```

See `a2png`, `antiword